o Don't loop forever if read() returns -1 due to some error.
--- Linux-PAM-0.77/libpam_misc/misc_conv.c	2003-06-30 22:28:29.000000000 -0400
+++ Linux-PAM-0.77/libpam_misc/misc_conv.c	2003-06-30 22:30:37.000000000 -0400
@@ -216,6 +216,13 @@
 		    fprintf(stderr, "\n");
 		}
 		goto cleanexit;                /* return malloc()ed "" */
+	    } else if (nc == -1) {
+		D(("error reading input from the user: %s", strerror(errno)));
+		if (echo) {
+		    fprintf(stderr, "\n");
+		}
+		input = NULL;
+		goto cleanexit;                /* return NULL */
 	    }
 	}
     }
